### ISSUE TITLE
Creating DGLGraph from non-consecutive int nodes

### ISSUE THREAD

**Issue by sufeidechabei** [dmlc/dgl#334]
## To Reproduce

```
    train_remove_length = len(np.where(data.train_mask== 1)[0]) * 0.1
    valid_remove_length = len(np.where(data.val_mask==1)[0]) * 0.1
    test_remove_length = len(np.where(data.test_mask== 1)[0]) * 0.1
    remove_train = np.random.choice(range(140), int(train_remove_length), replace=False)
    remove_valid = np.random.choice(range(200, 500), int(valid_remove_length), replace=False)
    remove_test = np.random.choice(range(500, 1500), int(test_remove_length), replace=False)
    total_delete_list = list(remove_train) + list(remove_valid) + list(remove_test)
    data.graph.remove_nodes_from(total_delete_list)
    graph = DGLGraph(data.graph)
```
```
dgl._ffi.base.DGLError: Error while creating graph from input of type "<class 'networkx.classes.digraph.DiGraph'>".
```

## Expected behavior

networkx graph to dgl graph

## Environment

 - DGL Version (e.g., 1.0): 0.1.3
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): Pytorch 0.4.1
 - OS (e.g., Linux): Linux
 - How you installed DGL (`conda`, `pip`, source): pip
 - Python version:3.6.7
 - CUDA/cuDNN version (if applicable): 9.0
 - GPU models and configuration (e.g. V100): V100
 - Any other relevant information:networkx(2.2)

**Comment by mufeili** [comment]
When you perform `data.graph.remove_nodes_from(total_delete_list)`, the nodes left will not be re-indexed. In `dgl` we are always indexing nodes with `0, ..., num_nodes-1`. Therefore you can end up asking to add edges between nodes dgl consider not to exist. A temporary workaround can be done by re-indexing the nodes and maintaining a mapping between the original node index and the node index after re-indexing.

**Comment by jermainewang** [comment]
We could provide a utility function similar to networkx's [`convert_node_labels_to_integers`](https://networkx.github.io/documentation/stable/reference/generated/networkx.relabel.convert_node_labels_to_integers.html#networkx.relabel.convert_node_labels_to_integers), and give a better error message for this.

**Comment by jermainewang** [comment]
@mufeili would you help on this?

**Comment by mufeili** [comment]
Sure.

**Comment by mufeili** [comment]
Fixed in #341
